App.tsx

```diff
@@ -1,17 +1,30 @@
 import * as SplashScreen from 'expo-splash-screen'
 import 'react-native-reanimated'
-import AppNavigator from './navigation'
-import { AuthProvider } from './contexts/AuthContext'
+import AppNavigator from "@/navigation";
+import { AuthProvider } from "@/contexts/AuthContext";
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
 import { StatusBar } from 'expo-status-bar'
 import { GestureHandlerRootView } from "react-native-gesture-handler";
+import { useFont } from "@/hooks/useFont";
+import { useEffect } from "react";
 
 // Prevent the splash screen from auto-hiding before asset loading is complete.
 SplashScreen.preventAutoHideAsync();
 
 const queryClient = new QueryClient();
 
 export default function App() {
+  const { loaded, error } = useFont();
+
+  useEffect(() => {
+    if (loaded || error) {
+      SplashScreen.hideAsync();
+    }
+  }, [loaded, error]);
+
+  if (!loaded && !error) {
+    return null;
+  }
```

Comment:
Am I missing something here? if the font is not loaded and there is not error we don't return the provider?

Reply:
Yes, basically there will be the splash screen in the meantime. remember that we have this line at the top `SplashScreen.preventAutoHideAsync();`

so when the font is not loaded yet or there is no error, there is a splash screen, and once it's loaded or there is an error the splash screen will be hidden, and then return the provider. 

so here we are making sure that there is no inconsistency in styling and flashes of fonts getting replaced